tests/sentry/logging/test_logging_exceptions.py

```diff
@@ -0,0 +1,38 @@
+import logging
+from unittest.mock import Mock
+
+import pytest
+import structlog
+from django.conf import settings
+from django.test import override_settings
+
+from sentry.logging.handlers import JSONRenderer, StructLogHandler
+from sentry.testutils.cases import TestCase
+
+
+class TestExceotionsDuringLogging(TestCase):
+    """
+    Test that exceptions during logging are handled gracefully in Production,
+    while still being surfaced in non-Production environments.
+    """
+
+    def setUp(self):
+        super().setUp()
+        structlog.configure(processors=[JSONRenderer()])
+        self.logger = logging.getLogger(__name__)
+        self.logger.addHandler(StructLogHandler())
```

Comment:
additionally these also set global variables that aren't undone -- they should be cleaned up outside your test

I'd recomend rewriting this using pytest where you have better control over teardown via yield fixtures